### PR TITLE
Improve HTCondor example and integration tests

### PR DESCRIPTION
**Pull request by tpdownes**
- Convert HTCondor example to multigroup deployment with custom image
- Modify integration test to use deploy/destroy commands
- Modify example to autoscale 2 idle nodes and add a test to check
- Increase yum/dnf timeout to 300 seconds to mitigate race condition with yum-cron/dnf-automatic installation of packages during boot

### Submission Checklist

Please take the following actions before submitting this pull request.

* Fork your PR branch from the Toolkit "develop" branch (not main)
* Test all changes with pre-commit in a local branch [#](https://goo.gle/hpc-toolkit#development)
* Confirm that "make tests" passes all tests
* Add or modify unit tests to cover code changes
* Ensure that unit test coverage remains above 80%
* Update all applicable documentation
* Follow Cloud HPC Toolkit Contribution guidelines [#](https://goo.gle/hpc-toolkit-contributing)

### CODE REVIEW

File: tools/cloud-build/daily-tests/ansible_playbooks/htcondor-integration-test.yml [hunk 1]
```diff
@@ -66,9 +62,12 @@
     - name: Get Builder IP
       register: build_ip
       changed_when: false
-      ansible.builtin.shell: >-
-        dig TXT +short o-o.myaddr.l.google.com @ns1.google.com |
-         awk -F'"' '{print $2}'
+      args:
+        executable: /bin/bash
+      ansible.builtin.shell: |
+        set -e -o pipefail
```
Comment (nick-stroud) [inline tools/cloud-build/daily-tests/ansible_playbooks/htcondor-integration-test.yml:68]: This represents drift from other tests. Options:
1. It seems unrelated to your change, so remove it
2. Update all occurrences (ex: base-integration-test.yml L96)
3. Move this into its own task (`tools/cloud-build/daily-tests/ansible_playbooks/tasks`) (prefered)

File: tools/cloud-build/daily-tests/ansible_playbooks/test-htcondor-access-point.yml [hunk 2]
```diff
@@ -13,10 +13,17 @@
...
   changed_when: False
-  until: ansible_fqdn == schedd_status.stdout
+  until: schedd_status.stdout == ansible_fqdn
   retries: 8
   delay: 15
+- name: Ensure execute points have joined the pool
+  ansible.builtin.command: condor_status -collector -autoformat HostsTotal
+  register: hosts_total
+  changed_when: False
+  until: hosts_total.stdout == "2"
```
Comment (nick-stroud) [inline tools/cloud-build/daily-tests/ansible_playbooks/test-htcondor-access-point.yml:27]: IIUC this magic number is coming from `community/examples/htc-htcondor.yaml` L122. Can we do more to connect the dots here. One idea would be turn this into a variable which would at least allow you to name it and default to 2. The name could be something like `htcondor_execute_point_min_idle_nodes`. In absence of that at least a comment.

**Review by nick-stroud — CHANGES_REQUESTED** [review]

File: community/examples/htc-htcondor.yaml [hunk 3]
```diff
@@ -17,19 +17,17 @@ blueprint_name: htc-htcondor
...
-    family: hpc-rocky-linux-8
+  disk_size_gb: 100
+  image_family: htcondor-10x
 
 # Documentation for each of the modules used below can be found at
 # https://github.com/GoogleCloudPlatform/hpc-toolkit/blob/main/modules/README.md
 
 deployment_groups:
-- group: htcondor
+- group: primary
```
[the remark this answers is not in the document]
Comment (tpdownes) [reply community/examples/htc-htcondor.yaml:30]: I generally do not like `primary` but would like to see our integration tests more broadly refactored to not assume its presecne before changing this (even in a multigroup deployment).

File: community/examples/htc-htcondor.yaml [hunk 4]
```diff
@@ -39,6 +37,28 @@ deployment_groups:
...
+  modules:
+  - id: custom-image
+    source: modules/packer/custom-image
+    kind: packer
+    use:
+    - network1
+    - htcondor_install_script
+    settings:
+      disk_size: $(vars.disk_size_gb)
+      source_image_family: hpc-rocky-linux-8
```
[the remark this answers is not in the document]
Comment (tpdownes) [reply community/examples/htc-htcondor.yaml:57]: The HPC VM Image project is one of the default projects searched by the `googlecompute` Packer plugin.

[the remark this answers is not in the document]
Comment (tpdownes) [reply community/examples/htc-htcondor.yaml:57]: https://github.com/hashicorp/packer-plugin-googlecompute/pull/120

File: community/examples/htc-htcondor.yaml [hunk 5]
```diff
@@ -17,19 +17,17 @@ blueprint_name: htc-htcondor
...
-  deployment_name: htcondor-001
+  deployment_name: htcondor-pool
   region: us-central1
   zone: us-central1-c
-  # Only CentOS 7 and Rocky 8 are supported; for CentOS 7, remove next 2 lines
-  instance_image:
-    project: cloud-hpc-image-public
-    family: hpc-rocky-linux-8
+  disk_size_gb: 100
+  image_family: htcondor-10x
```
[the remark this answers is not in the document]
Comment (tpdownes) [reply community/examples/htc-htcondor.yaml:24]: Done

File: community/examples/htc-htcondor.yaml [hunk 6]
```diff
@@ -17,19 +17,17 @@ blueprint_name: htc-htcondor
 
 vars:
   project_id:  ## Set GCP Project ID Here ##
-  deployment_name: htcondor-001
+  deployment_name: htcondor-pool
   region: us-central1
   zone: us-central1-c
-  # Only CentOS 7 and Rocky 8 are supported; for CentOS 7, remove next 2 lines
```
[the remark this answers is not in the document]
Comment (tpdownes) [reply community/examples/htc-htcondor.yaml:23]: Adding a modified section of `htcondor-install` README to address this concern.

File: tools/cloud-build/daily-tests/ansible_playbooks/htcondor-integration-test.yml (hunk 1 above)
Comment (tpdownes) [reply tools/cloud-build/daily-tests/ansible_playbooks/htcondor-integration-test.yml:68]: Selecting option (2) as a dedicated commit.

**Review by nick-stroud — APPROVED** [review]
